### PR TITLE
fixes for NEXGEN

### PR DESCRIPTION
**Pull request by sglienke**

### CODE REVIEW

File: DUnitX.WeakReference.pas [hunk 1]
```diff
@@ -62,23 +66,26 @@ interface
...
     FWeakReferences : TList<Pointer>;
-    FDestroying : boolean;
-    FDestroyed : boolean;
     function QueryInterface(const IID: TGUID; out Obj): HResult; stdcall;
     function _AddRef: Integer; stdcall;
     function _Release: Integer; stdcall;
     procedure AddWeakRef(value : Pointer);
     procedure RemoveWeakRef(value : Pointer);
-    function GetRefCount : integer;
+    function GetRefCount : integer; inline;
```
Comment (vincentparrett) [inline DUnitX.WeakReference.pas]: Shouldn't this method be ifdef's as well? It uses FRefCount which is ifdef out?

**Comment by vincentparrett** [comment]
Looks good, apart from one small issue (see inline comment). It compiles fine and passes the tests on my build server for win32/win64. I will test this code in FinalBuilder as it's a complex app with a lot of use of the weak ref, see if it throws up any issues.

File: DUnitX.WeakReference.pas (hunk 1 above)
Comment (sglienke) [reply DUnitX.WeakReference.pas]: It uses FRefCount from TObject under NEXTGEN.

Comment (vincentparrett) [reply DUnitX.WeakReference.pas]: Ah right, I didn't realise it was protected, I thought it was private.

**Pull request merged**